### DISCUSSION TITLE
Issue with HexagonMeshTrimmer After Using BlockDeletionGenerator

### DISCUSSION THREAD

**Discussion by hityyds** [idaholab/moose#29089] [Q&A Meshing] [answered]
Hello,

I encountered an issue when using HexagonMeshTrimmer in MOOSE after applying BlockDeletionGenerator to delete the dummy blocks. The following error message is displayed:
`The input mesh is not compatible with HexagonMeshTrimmer because the trimmability mesh meta data are absent.`

Here is the sequence of steps I followed to generate the mesh:
1. Used PolygonConcentricCircleMeshGenerator to create pins.
2. Used PatternedHexMeshGenerator to create assemblies.
3. Applied PatternedHexPeripheralModifier to adjust assemblies.
4. Used HexagonConcentricCircleAdaptiveBoundaryMeshGenerator to generate dummies.

When I use HexagonMeshTrimmer on this mesh without applying BlockDeletionGenerator, everything works as expected. However, when I use BlockDeletionGenerator before HexagonMeshTrimmer (with everything else remaining the same), the error above occurs.

I’ve reviewed the documentation for HexagonMeshTrimmer and meshing tutorials, and it seems like my usage should be correct. Could you help me understand why this happens and how to resolve it?

Thank you for your assistance!

**Comment by GiudGiud** [comment, marked as answer]
Hello

The HexagonMeshTrimmer requires a boolean metadata to be sure that the mesh is trimmable. 
Using a BlockDeletionGenerator removes the metadata because we can no longer guarantee that the mesh is trimmable if you use custom operations on it. 
If you wish to re-instate this metadata you can use:
https://mooseframework.inl.gov/source/meshgenerators/AddMetaDataGenerator.html

to set the `hexagon_peripheral_trimmability` and `hexagon_center_trimmability` booleans yourself. Center trimmability means you can cut some sectors off without breaking elements in the mesh, and similarly for peripheral trimmability, it means the periphery of the mesh can be removed. If the mesh is not truly trimmable, then your output mesh will have jagged edges. 

Guillaume

**Reply by GiudGiud** [reply]
using the BlockDeletionGenerator after the HexTrimmer is the better solution here

**Reply by hityyds** [reply]
Thank you very much for your help! I’ll try your suggestion to use AddMetaDataGenerator to address the issue.

In my case, I need to first delete the dummy assemblies and then use PeripheralRingMeshGenerator to add a cylindrical peripheral to the mesh. Finally, I apply HexagonMeshTrimmer. This sequence is essential because changing the order of HexagonMeshTrimmer and BlockDeletionGenerator would leave a hole in the middle of the mesh, which is not acceptable for my use case.

Thanks again for your guidance!
